src/bp/ui-studio/src/web/views/OneFlow/diagram/SubworkflowForm/InputForm.tsx

```diff
@@ -0,0 +1,79 @@
+import { Tab, Tabs } from '@blueprintjs/core'
+import { FlowVariable, FormField, SubWorkflowInput } from 'botpress/sdk'
+import { Contents, lang, RightSidebar } from 'botpress/shared'
+import { Variables } from 'common/typings'
+import _ from 'lodash'
+import React, { FC, Fragment, useEffect, useState } from 'react'
+
+import style from '../PromptForm/style.scss'
+
+interface Props {
+  customKey: string
+  subFlowVars: FlowVariable[]
+  formData: { [variable: string]: SubWorkflowInput }
+  variables: Variables
+  close: () => void
+  updateEntry: (formData: any) => void
+  onUpdateVariables: (variable: FlowVariable) => void
+}
+
+const InputForm: FC<Props> = ({
+  close,
+  customKey,
+  formData,
+  subFlowVars,
+  variables,
+  updateEntry,
+  onUpdateVariables
+}) => {
+  const [fieldTypes, setFieldTypes] = useState({})
+
+  useEffect(() => {
+    setFieldTypes(_.mapValues(formData, item => (item?.source === 'variable' ? 'variable' : 'text')))
+  }, [customKey])
+
+  const fields = subFlowVars.map<FormField>(({ params, type }) => ({
+    type: fieldTypes[params.name] ?? 'variable',
+    key: params.name,
+    label: params.name,
+    variableTypes: [type],
+    placeholder: 'module.builtin.setValueToPlaceholder',
+    defaultVariableType: type,
+    onClick: field => setFieldTypes({ ...fieldTypes, [field.key]: field.type === 'variable' ? 'text' : 'variable' })
```

Comment:
Can you put a TODO here to use the variable's default Field type instead of `text` ? For example if the variable is a date, we'll have a date picker

Reply:
Yes sure